### DISCUSSION TITLE
Bundler/bundle not using cache in Github Actions

### DISCUSSION THREAD

**Discussion by avimoondra** [community/community#25350] [Actions]
<p>With the following configuration…</p>
<pre><code class="lang-auto">  test-jekyll:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-ruby@v1
        with:
          ruby-version: "2.6"
      - uses: actions/cache@v2
        with:
          path: vendor/bundle
          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
          restore-keys: |
            ${{ runner.os }}-gems-
      - run: gem install bundler
      - run: bundle config path vendor/bundle
      - run: bundle install --jobs 4 --retry 3
      - run: mkdir public
      - run: bundle exec jekyll build --destination ./public
        env:
          JEKYLL_ENV: production
      - name: Deploy
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_branch: gh-pages-test
          enable_jekyll: true
</code></pre>
<p>Seems to hit the cache file, but not use the cache when running bundle install. It ends up taking 3+ minutes!  I’m following the same instructions as here:  <a href="https://github.com/actions/cache/blob/ff937cc95032836a535d249de4ce2fc52aeae834/examples.md#ruby---bundler" rel="nofollow noopener">https://github.com/actions/cache/blob/ff937cc95032836a535d249de4ce2fc52aeae834/examples.md#ruby---bundler</a></p>
<aside class="onebox githubpullrequest">
  <header class="source">
      <a href="https://github.com/defund12/defund12.org/pull/1369/checks?check_run_id=760862303" target="_blank" rel="nofollow noopener">github.com/defund12/defund12.org</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/defund12/defund12.org/pull/1369" target="_blank" rel="nofollow noopener">[do not merge] test new jekyll flow</a>
    </h4>

    <div class="branches">
      <code>defund12:master</code> ← <code>defund12:avimoondra/test-new-jekyll-flow</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-06-11" data-time="06:29:32" data-timezone="UTC">06:29AM - 11 Jun 20 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/avimoondra" target="_blank" rel="nofollow noopener">
          <img alt="avimoondra" src="https://user-images.githubusercontent.com/2534903/181090568-02d072b8-ea48-4350-a0f4-5469c5e7f79d.png" class="onebox-avatar-inline" width="20" height="20">
          avimoondra
        </a>
      </div>

      <div class="lines" title="1 commits changed 1 files with 26 additions and 0 deletions">
        <a href="https://github.com/defund12/defund12.org/pull/1369/files" target="_blank" rel="nofollow noopener">
          <span class="added">+26</span>
          <span class="removed">-0</span>
        </a>
      </div>
    </div>

  </div>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>As a live example^^</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/1c5bad854d6d964bcc6e0bc9705895c9baa62e8e.png" data-download-href="/uploads/short-url/42RMR175XqvjVVPSefnsOB1ME22.png?dl=1" title="Screen Shot 2020-06-11 at 10.37.07 AM" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2534903/181090569-174fe97e-6fe4-4a74-b61d-0447f202d070.png" alt="Screen Shot 2020-06-11 at 10.37.07 AM" data-base62-sha1="42RMR175XqvjVVPSefnsOB1ME22" width="690" height="278" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/1c5bad854d6d964bcc6e0bc9705895c9baa62e8e_2_690x278.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/1c5bad854d6d964bcc6e0bc9705895c9baa62e8e_2_1035x417.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/1/1c5bad854d6d964bcc6e0bc9705895c9baa62e8e_2_1380x556.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/1c5bad854d6d964bcc6e0bc9705895c9baa62e8e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-06-11 at 10.37.07 AM</span><span class="informations">3128×1264 304 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Is there a configuration I need to change to have <code>bundle install</code> command use the cache?</p>
<p>Thanks!</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/avimoondra">@avimoondra</a>,</p>
<p>Thank you for being here!<br>
In your workflow, cache hit cause bundle installation is not stored into cache. Hence, it will take 3mins to install again. You can edit Gemfile.lock and push an empty submit, change the hash value will cause a cache miss, new cache will contain your bundle installation content. Then, you will find if cache hit next time, it will reduce much of time for the step.</p>
<p>I forked your repo, if i manually create a catch miss in prior workflow, in next workflow, it only takes 2s for the step.<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/weide-zhou/defund12.org/runs/764620953?check_suite_focus=true#step:8:1" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181090571-72700408-a6ff-4b40-bd82-49063edf8191.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/weide-zhou/defund12.org/runs/764620953?check_suite_focus=true#step:8:1" target="_blank" rel="nofollow noopener">weide-zhou/defund12.org</a></h3>

<p>defund12.org. Contribute to weide-zhou/defund12.org development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Thanks.</p>

**Comment by avimoondra** [comment]
<p>Great this worked - had to bust the cache! Thanks <a class="mention" href="/weide-zhou">@weide-zhou</a></p>
